src/MonteCarlo/ParallelTempering.jl

```diff
@@ -1,27 +1,42 @@
 # vanilla parallel tempering using Julia multithreading
-mutable struct ParallelTempering
+mutable struct ParallelTempering{N, SMP}
```

Comment:
Side-comment -- type parameters like {N} and {SMP} are useful for describing intent, and also useful for getting maximum performance, but may have the downside of creating more compilation work, and defeating precompilation. At some point we should go through all of Sunny and try to figure out whether it's possible to remove type parameters. For example, in this context, it may not be performance critical to have `N` statically known. To figure out it would require a lot of benchmarking.